### ISSUE TITLE
` Layout/MultilineMethodCallIndentation` `indented` breaks down with method call before assignment

### ISSUE THREAD

**Issue by AlexWayfer** [rubocop/rubocop#10456]
## Expected behavior

No offenses from `Layout/MultilineMethodCallIndentation`.

## Actual behavior

Offense per each line from `Layout/MultilineMethodCallIndentation`.

```
For /home/alex/Projects/ruby/test/rubocop/method_chaining: configuration from /home/alex/Projects/ruby/test/rubocop/method_chaining/.rubocop.yml
Default configuration from /home/alex/.rbenv/versions/3.1.1/lib/ruby/gems/3.1.0/gems/rubocop-1.26.0/config/default.yml
AllCops/Exclude configuration from /home/alex/Projects/ruby/test/.rubocop.yml
Use parallel by default.
Running parallel inspection
Inspecting 1 file
Scanning /home/alex/Projects/ruby/test/rubocop/method_chaining/test.rb
Loading cache from /home/alex/.cache/rubocop_cache/4c137b0d5f5d8fdbc0e9bb4b9f6dd0b1df7fe331/2d086c9a02b07de22b92121fa4e42b028afffdec/7d4ab545ae940e92d10de1093604eedb0b3bd020
C

Offenses:

test.rb:5:5: C: [Correctable] Layout/MultilineMethodCallIndentation: Use 2 (not 4) spaces for indenting an expression in an assignment spanning multiple lines.
    .gsub(/\s+(?=$)/, '')
    ^^^^^
test.rb:6:5: C: [Correctable] Layout/MultilineMethodCallIndentation: Use 2 (not 4) spaces for indenting an expression in an assignment spanning multiple lines.
    .gsub(/([,;]|\.{2,})(?=$)/, '.')
    ^^^^^

1 file inspected, 2 offenses detected, 2 offenses auto-correctable
Finished in 0.13953729800050496 seconds
```

## Steps to reproduce the problem

```yaml
# .rubocop.yml

AllCops:
  NewCops: enable

Layout/MultilineMethodCallIndentation:
  EnforcedStyle: indented
```

```rb
# frozen_string_literal: true

self.bar =
  baz
    .gsub(/\s+(?=$)/, '')
    .gsub(/([,;]|\.{2,})(?=$)/, '.')
```

(just removing `self.` leads to no offenses)

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
1.26.0 (using Parser 3.1.1.0, rubocop-ast 1.16.0, running on ruby 3.1.1 x86_64-linux)
```

**Comment by AlexWayfer** [comment]
@koic thank you, I appreciate your work! :raised_hands:
